docs/cip/CIP-8_Matrix_Packages.md

```diff
@@ -0,0 +1,209 @@
+
+Exactly how to apply this principle needs consideration for each section.
+
+TODO how much of this can or should be encoded directly in the parquet file? is duplication OK? what would be the reason for having it here instead of or in addition to parquet?
+
+##### `data.embedding`
+```yaml
+  embedding:
+    model: "hf://sentence-transformers/all-MiniLM-L6-v2"
+    types:
+      dimensions: 384
+      quantize:
+        bits: 8
+        method: "k-means"
+```
+
+The purpose of this stanza is to describe type-like properties of the vector space for the embeddings in the package. It is an attempt at "where, not how."
+
+`embedding.model` tells us the model that produced the embeddings in the data file. A simple string identifier for the model that produced the embeddings is sufficient here to establish what amounts to a [nominal type system](https://medium.com/@thejameskyle/type-systems-structural-vs-nominal-typing-explained-56511dd969f4). The name of the model here can then be thought of as the name of the universe of weights defined by the trained model, and by extension the computed embeddings.
+
+There is plenty of debate on the virtues of structural vs. nominal typing in the traditional software world. It often amounts to, "is it better DX to assume that 'if it quacks like a duck, it's a `Duck`' (structural), or to require explicit opt-in on a definition of `Duck`? (nominal)" Nominal type systems also implicitly allow names to carry opaque, special meaning beyond their formally expressed properties. That opacity makes nominal typing the natural choice for this case, as it mirrors the opacity of a trained model's weights.
+
+`embedding` is closely related to, but more constrained than Chroma's embedding function - it names only the model, rather some known invokable function whose (opaque) behavior is to call that model.
+
+TODO is this over-abstracting the notion of an embedding func, to the point where it's not useful at all?
+
+`embedding.types` describes other knowable properties about the vector space in which the embeddings live - what's the dimensionality of the space? is it quantized? For Chroma's purposes, these descriptive properties should be treated as additional constraints on the [query pipeline](https://github.com/chroma-core/chroma/blob/6384b66bc19e6a6217c2c5aa55f3b539e30c2708/docs/CIP_6_Pipelines_Registry.md#query-pipelines) that must be applied when processing an input for vector search.
+
+TODO i am totally out of my depth with quantization and need to run this through with someone. Key questions - does it make sense to try to reduce embedding and query pipeline fns to a model + some set of constraint arguments? Is quantization distinct from embedding - i.e. do we have one column for each in the parquet file?
+TODO finish
+
+#### `data.digests`
+```yaml
+  digests:
```

Comment:
If OCI is the target format, don't we want to keep these in the manifest?

```json
{
  "schemaVersion": 2,
  "config": {
    "mediaType": "application/vnd.chroma.config.v1+yaml",
    "digest": "sha256:abc123",
    "size": 7023,
    "annotations": {
      "org.opencontainers.image.title": "matrix.package.yaml"
    }
  },
  "layers": [
    {
      "mediaType": "application/vnd.chroma.layer.v1+parquet",
      "digest": "sha256:def456",
      "size": 32654,
      "annotations": {
        "org.opencontainers.image.title": "data.parquet"
      }
    }
  ]
}
```

Reply:
Yeah, it certainly seems like we need to at least be able to produce that information from `matrix.package.yaml`.

While the document leans hard on OCI registries, and i have certainly been a consumer of them, i've never actually used them as the backing store for something i've implemented. (i nevertheless feel OK leaning on them in the doc because of the sheer number of folks i know and respect who've done so)

Clearly it's possible to do some kind of layering here - dockerfiles don't look like that, but OCI registries were originally designed to satisfy the docker layered fs case - but i don't know how that works. That's a TODO to figure out, as if possible i'd rather avoid creating confusion and footguns by exposing what feels like internal details in `matrix.package.yaml`

So, i opted to include just the hash digest